spec/controllers/v2/concerns/with_metadata_spec.rb

```diff
@@ -35,4 +38,47 @@
 
     expect(response.parsed_body['meta']['total']).to eq(item_count)
   end
+
+  it 'has no previous link when on first offset' do
+    items
+
+    get :index, params: extra_params.merge(
+      limit: 2,
+      offset: 0,
+      parents: parents
+    )
+
+    expect(response.parsed_body['links'].keys).to contain_exactly(*first_offset_keys)
+  end
+
+  it 'has all four links when not on first or last offset' do
+    items
+
+    get :index, params: extra_params.merge(
+      limit: 2,
+      offset: 2,
+      parents: parents
+    )
+
+    expect(response.parsed_body['links'].keys).to contain_exactly(*middle_offset_keys)
+  end
+
+  it 'has no next link when on last offset' do
+    items
+
+    get :index, params: extra_params.merge(
+      limit: 10,
+      offset: 9,
+      parents: parents
+    )
+
+    expect(response.parsed_body['links'].keys).to contain_exactly(*last_offset_keys)
+  end
+
+  it 'has correct last offset when returning not_found' do
+    get :index, params: extra_params.merge(parents: parents)
+
+    expect(response_body_data).to be_empty
+    expect(response.parsed_body['links']['last'][-1]).to eq('0')
```

Comment:
I think this needs some explanation in a comment what exactly is happening here. The `it` block above says "correct", but what does it actually mean here?